### PR TITLE
Prepare for release of new crate

### LINKED ISSUE #3: Unwind safety issues for Kotlin and Swift bindings
```rust
#[no_mangle]
pub unsafe extern "C" fn stretch_node_compute_layout(
    stretch: *mut c_void,
    node: *mut c_void,
    width: f32,
    height: f32,
    create_layout: fn(*const f32) -> *mut c_void,
) -> *mut c_void {
    let mut stretch = Box::from_raw(stretch as *mut Stretch);
    let node = Box::from_raw(node as *mut Node);

    stretch
        .compute_layout(
            *node,
            Size {
                width: if f32::is_nan(width) { Number::Undefined } else { Number::Defined(width) },
                height: if f32::is_nan(height) { Number::Undefined } else { Number::Defined(height) },
            },
        )
        .unwrap();

    let mut output = vec![];
    copy_output(&stretch, *node, &mut output);

    Box::leak(stretch);
    Box::leak(node);

    create_layout(output.as_ptr())
}
```

The unwinding of `stretch.compute_layout` will introduce double free due to dropping `stretch`  and `node` in the unwinding path. With the appearance of this question, we noticed that this kind of `from_raw` usage is not strong enough for FFI functions😨.

The `leak` should take forward to avoid the automated deallocation towards objects generating by `from_raw`. We found `std::mem::ManuallyDrop` is a good wrapper for us to avoid the drop as well as making the variable act like the smart pointer.

The fixed code we listed below. And this refinement is zero-cost.

```rust
#[no_mangle]
pub unsafe extern "C" fn stretch_node_compute_layout(
    stretch: *mut c_void,
    node: *mut c_void,
    width: f32,
    height: f32,
    create_layout: fn(*const f32) -> *mut c_void,
) -> *mut c_void {
    let mut stretch = mem::ManuallyDrop::new(Box::from_raw(stretch as *mut Stretch));
    let node = mem::ManuallyDrop::new(Box::from_raw(node as *mut Node));

    stretch
        .compute_layout(
            *node,
            Size {
                width: if f32::is_nan(width) { Number::Undefined } else { Number::Defined(width) },
                height: if f32::is_nan(height) { Number::Undefined } else { Number::Defined(height) },
            },
        )
        .unwrap();

    let mut output = vec![];
    copy_output(&stretch, *node, &mut output);
    create_layout(output.as_ptr())
}
```
We think all the usage (involving 24 functions) of `from_raw` in bindings for Swift and Kotlin should be changed by wrapping with `ManuallyDrop` to avoid double free incurred by unwinding 😊.

### LINKED ISSUE #11: Consider renaming the crate
stretch2 is not exactly a compelling name.

The sooner we rebrand, the less confusing it will be.

### LINKED ISSUE #24: Build binary files in CI
Generated bindings require the inclusion of binary code, as seen in #4. This is effectively impossible to audit, and a pain for contributors and maintainers.

These should be built automatically in some form instead.

### LINKED ISSUE #25: Automate and host integrations for other languages
This was done by stretch originally, but relied on their own proprietary infrastructure.

We should either remove these bits of the code base, or navigate the JS / Kotlin / Swift / other ecosystems to do this properly again.

I have no personal need for these integrations: we should try to onboard maintainers who do care if we want to keep this functionality.

### LINKED ISSUE #38: Remove `.circleci` folder
We don't use circle-ci, so we can probably just remove that folder.

### PR DESCRIPTION
**Pull request by alice-i-cecile**
# Objective

- Rename the crate.
  - Fixes #11.
- Remove JS / Swift / Kotlin bindings.
  - Closes #24.
  - Closes #25.
  - Closes #3.
- Remove unused .circleci config.
  - Closes #38.
- Rewrite README to reflect project direction.

## Context

Using the current leader in #42. If something else wins, we can do a simple find-and-replace.

For reviewers: review this on a commit-by-commit basis. https://github.com/DioxusLabs/stretch/pull/53/commits/32bbc330f561ae63a2e253cb5c1d621d638d983e is a massive but trivial find-and-replace on the generated tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,130 +19,16 @@ Before using or contributing to stretch2 it is good to be aware of the core goal
...
-const node = new Node(allocator, {width: 100, height: 100, justifyContent: JustifyContent.Center});
-node.addChild(new Node(allocator, {width: '50%', height: '50%'}));
-const layout = node.computeLayout();
-
-console.log(layout.width, layout.height);
-```
-
-### Installation
-
 If you don't have Rust installed you have to do that first as well as install some components that we make use of to format and lint the codebase. For more on Rust see their [website](https://www.rust-lang.org).
```
Comment (TimJentzsch) [inline README.md]: This is a very vague description that could be improved. We could link directly to the Rust installer here (<https://www.rust-lang.org/tools/install>).

File: Cargo.toml [hunk 2]
```diff
@@ -4,9 +4,9 @@ version = "0.4.2"
 authors = ["Alice Cecile <alice.i.cecile@gmail.com>", "Johnathan Kelley <jkelleyrtp@gmail.com>"]
 edition = "2021"
 include = ["src/**/*", "Cargo.toml"]
-description = "High performance & cross-platform Flexbox implementation"
+description = "A flexible, well-crafted, high performance UI layout library"
```
Comment (TimJentzsch) [inline Cargo.toml:7]: Personally, I'm not a big fan of `well-crafted` as that is kinda subjective IMO.

File: RELEASES.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+### Changed
+
+- renamed crate from `stretch2` to `sprawl`
+
+### Removed
+
+- removed Javascript / Kotlin / Swift bindings
+  - the maintainer team lacks expertise to keep these working
+  - more serious refactors are planned, and this will be challenging to keep working through that process
+  - if you are interested in helping us maintain bindings to other languages, get in touch!
```
Comment (TimJentzsch) [inline RELEASES.md:14]: We could link the GitHub Discussions here (assuming the repo will be moved to DioxusLabs/sprawl):
```suggestion
  - if you are interested in helping us maintain bindings to other languages, [get in touch](https://github.com/DioxusLabs/sprawl/discussions)!
```

File: Cargo.toml (hunk 2 above)
Comment (TimJentzsch) [reply Cargo.toml:7]: (And I'm not sure if we can confidently call it well-crafted before doing some refactors :P)

**Comment by alice-i-cecile** [comment]
@mockersf, any ideas on [what's going on with CI](https://github.com/alice-i-cecile/stretch2/runs/6528154954?check_suite_focus=true#step:5:9) here?

File: Cargo.toml (hunk 2 above)
Comment (alice-i-cecile) [reply Cargo.toml:7]: High performance is subjective too :p I'd like to convey a sense of "high code-quality, good docs, solid tests".

Cutting for now though.

Comment (alice-i-cecile) [reply Cargo.toml:7]:
```suggestion
description = "A flexible high performance UI layout library"
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -5,13 +5,35 @@ New contributions are extremely welcome!
...
-1. Pick an [issue](https://github.com/DioxusLabs/stretch/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22), or [file a new one](https://github.com/DioxusLabs/stretch/issues/new).
+1. Pick an [issue](https://github.com/DioxusLabs/sprawl/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22), or [file a new one](https://github.com/DioxusLabs/sprawl/issues/new).
 2. Comment in the issue that you plan to tackle it, and the team will assign the task to you.
 3. Submit a PR.
 4. Respond to feedback from reviewers and make sure CI passes.
 
 The PR review process is completely open:  help us by commenting on, testing and approving PRs.
 
-Questions like "how do I use this library?" belong in [GitHub Discussions](https://github.com/DioxusLabs/stretch/discussions/new), although those should typically spawn new issues to improve our docs and examples.
+Questions like "how do I use this library?" belong in [GitHub Discussions](https://github.com/DioxusLabs/sprawl/discussions/new), although those should typically spawn new issues to improve our docs and examples.
```
Comment (alice-i-cecile) [inline CONTRIBUTING.md:15]:
```suggestion
```

File: README.md (hunk 1 above)
Comment (alice-i-cecile) [reply README.md]: This section has been completely removed. I agree, it's not very useful.

**Comment by TimJentzsch** [comment]
> @mockersf, any ideas on [what's going on with CI](https://github.com/alice-i-cecile/stretch2/runs/6528154954?check_suite_focus=true#step:5:9) here?

To me it looks like it's trying to load something in `stretch2`, but can't find it due to the crate rename in `Cargo.toml`? Not entirely sure though.

**Comment by alice-i-cecile** [comment]
Yeah, it feels like it might be caching something?

**Comment by mockersf** [comment]
you deleted the folder `scripts/gentest`, but it's still present in the workspace definition

**Comment by TimJentzsch** [comment]
> you deleted the folder scripts/gentest, but it's still present in the workspace definition

I wonder, what exactly is the difference between `scripts/gentest` and `tests/generated`? Does the former generate the latter? And why are the generated files committed to the repo? For CI?

**Comment by mockersf** [comment]
Yup and yup. If I remember correctly the tests are quite heavy to generate and uses a webbrowser, to generate the assertion values

**Comment by alice-i-cecile** [comment]
With the final release of stretch2, this is now unblocked. I'm going to merge this aggressively; further work should be done in easier-to-review PRs.

**Pull request merged**